MapboxDirections/MBDirectionsRouter.swift

```diff
@@ -48,83 +49,65 @@ internal enum MBDirectionsRouter: Router {
         case Full = "full"
     }
     
-    case V4(Configuration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
-    case V5(Configuration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
-    
-    var method: HTTPMethod {
-        return .GET
-    }
-    
-    var encoding: HTTPEncoding {
-        return .URL
-    }
-    
-    var configuration: Configuration {
-        switch self {
-        case .V4(let config, _, _, _, _, _, _): return config
-        case .V5(let config, _, _, _, _, _, _, _): return config
-        }
-    }
+    case V4(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, InstructionFormat?, GeometryFormatV4?, Bool?)
+    case V5(MBDirectionsConfiguration, String, [MBDirectionsWaypoint], Bool?, GeometryFormatV5?, OverviewGranularity?, Bool?, Bool?)
```

Comment:
Yeah, it's odd to switch v4/v5 everywhere - it seems weird to support more than one api version at a time, or if we do, we could treat them as separate services with totally different code paths?